### ISSUE TITLE
add annotation doesn't work as expected

### ISSUE THREAD

**Issue by chribsen** [kubernetes-sigs/kustomize#492]
I'm having issues with the `kustomize` CLI when trying to add an annotation to a `kustomization.yaml`. 

Upon running the following command, I get the following error:
```
$ kustomize edit add annotation {foo:bar}
Error: field: Invalid value: "{foo": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
```

It looks like `kustomize` is applying the validation regex `{foo` instead of just `foo`.

*Version:*
```
$ kubectl version
Version: {KustomizeVersion:v1.0.8 GitCommit:58492e2d83c59ed63881311f46ad6251f77dabc3 BuildDate:1970-01-01T00:00:00Z GoOs:unknown GoArch:unknown}
```

**Comment by Liujingfang1** [comment]
This issue was fixed by #482.

**Comment by Liujingfang1** [comment]
The command should be
`kusotmize edit add annotation foo:bar`

The doc for this part may need improvement.
